applications/vanilla/controllers/class.discussioncontroller.php

```diff
@@ -112,8 +112,6 @@ public function Index($DiscussionID = '', $DiscussionStub = '', $Page = '') {
             // (((67 comments / 10 perpage) = 6.7) rounded down = 6) * 10 perpage = offset 60;
             $this->Offset = floor($CountCommentWatch / $Limit) * $Limit;
          }
-         if ($ActualResponses <= $Limit)
-            $this->Offset = 0;
```

Comment:
I think this is incorrect. If there are less comments than the page size then the offset is always zero when in auto-offset. Moving this removed statement to within the previous if statement is the way to go.